ui/plugins/controller.js

```diff
@@ -187,24 +190,25 @@ treeherder.controller('PluginCtrl', [
             if (job.id) {
                 $scope.job_detail_loading = true;
                 if (selectJobCanceller !== null) {
```

Comment:
This is pre-existing (even before the recent #3570 change), however it looks like `selectJobCanceller` is never set back to `null` after the requests completes. 

However, looking at the MDN examples I think it should be fine to re-use the same `AbortController()` instance each time (presuming the `.abort()` works if used multiple times) - which means we can remove the conditional entirely. eg:

```js
const selectJobController = new AbortController();

const selectJob = function (job) {
  // ...
  if (job.id) {
    // Cancel any in-progress fetch requests.
    selectJobController.abort();
    $scope.job_detail_loading = true;
    // ...
};

$rootScope.$on(thEvents.clearSelectedJob, function () {
  // Cancel any in-progress fetch requests.
  selectJobController.abort();
});
```

I've also replaced `window.AbortController` with `AbortController` in that example. If the older ESLint version we're using errors (because its list of known browser globals is out of date), then lets add an eslint ignore line with a TODO to remove it once we're on a newer version :-)

Reply:
Ends up if we re-use the AbortController, it won't make the calls when you switch jobs.  Apparently it considers them "already aborted" before they even start.  So I had to re-create that object each time, like we were before.

Reply:
I added in:

    .finally(() => {
                    selectJobController = null;
                })

to null it out again though.

Reply:
Makes sense. Could you still rename to `selectJobController` and remove the `window.` :-)